internal/topo/context/decoder.go

```diff
@@ -37,3 +37,23 @@
 	}
 	return nil, fmt.Errorf("no decoder configured")
 }
+
+func (c *DefaultContext) DecodeIntoList(data []byte) ([]interface{}, error) {
+	v := c.Value(DecodeKey)
+	f, ok := v.(message.Converter)
+	if ok {
+		t, err := f.Decode(data)
+		if err != nil {
+			return nil, fmt.Errorf("decode failed: %v", err)
+		}
+		typeErr := fmt.Errorf("only map[string]interface{} and []map[string]interface{} is supported but got: %v", t)
+		if result, ok := t.(map[string]interface{}); ok {
+			return []interface{}{result}, nil
+		}
+		if result, ok := t.([]interface{}); ok {
```

Comment:
Maybe also allow []map[string]interface{}. Since there are more than 1 check, is it better to use type switch

Reply:
As we use `interface{}` for the `json.Unmarshal`. For the data below:
```json
[{"a":1},{"a":1}]
```
The converter will convert it into
```
[]interface{} {
 map[string]interface{} {"a":1}
 map[string]interface{} {"a":2}
}
```

Thus we don't assert `t` should be `[]map[string]interface{}` here

Reply:
JSON is just one of the decode method. Custom method may return []map

Reply:
updated.